apps/desktop-wallet/src/components/Inputs/Select.tsx

```diff
@@ -276,40 +270,40 @@ export function SelectOptionsModal<T extends OptionValue>({
   hookCoordinates,
   title,
   optionRender,
-  onSearchInput,
   searchPlaceholder,
-  showOnly,
   emptyListPlaceholder,
   parentSelectRef,
   minWidth,
   ListBottomComponent,
-  floatingOptions
+  floatingOptions,
+  isSearchable
 }: SelectOptionsModalProps<T>) {
   const { t } = useTranslation()
   const optionSelectRef = useRef<HTMLDivElement>(null)
   const theme = useTheme()
   const searchInputRef = useRef<HTMLInputElement>(null)
 
-  // We hide instead of simply not rendering filtered options to avoid changing the height/width of the modal when
-  // filtering. When the size of the modal depends on its contents, its size might change when filtering some options
-  // out, unless its size is fixed.
-  const [visibleOptions, invisibleOptions] = showOnly
-    ? partition(options, (option) => showOnly.includes(option.value))
-    : [options, []]
+  const [searchInput, setSearchInput] = useState('')
+  const filteredOptions =
+    searchInput.length > 2
+      ? options.filter((option) =>
+          option.searchString
+            ? option.searchString.includes(searchInput)
+            : option.label.toLocaleLowerCase().includes(searchInput)
+        )
+      : options
   const isEmpty = options.length === 0 && emptyListPlaceholder
-  const emptySearchResults = visibleOptions.length === 0 && onSearchInput
-  // To display the message without changing the height, remove one of the invisible options
-  if (emptySearchResults) invisibleOptions.pop()
+  const emptySearchResults = filteredOptions.length === 0 && isSearchable
```

Comment:
We are not filtering the options if `isSearchable` is `false` because then `searchInput` is `''`, which means that `filteredOptions` is the same as `options`!